### PR TITLE
Fix RandomSizedBBoxSafeCrop.

### PR DESCRIPTION
**Pull request by Jae-Hyuck**
To my understanding,  even though 'bboxes' is given,
'get_params_dependent_on_targets' method cannot return proper params,
because 'targets_as_params' does not include 'bboxes'.

So, I add 'bboxes' to the 'targets_as_params'.
Now, 'bboxes' is necessary to run RandomSizedBBoxSafeCrop. (An empty list is allowed.)

Please check my thinking is correct.
I also changed some test code.
Thank you.

### CODE REVIEW

File: albumentations/augmentations/transforms.py [hunk 1]
```diff
@@ -846,7 +846,7 @@ def apply(self, img, crop_height=0, crop_width=0, h_start=0, w_start=0, interpol
 
     def get_params_dependent_on_targets(self, params):
         img_h, img_w = params['image'].shape[:2]
-        if 'bboxes' not in params:  # less likely, this class is for use with bboxes.
+        if len(params['bboxes']) == 0:  # less likely, this class is for use with bboxes.
```
Comment (albu) [inline albumentations/augmentations/transforms.py]: if `bboxes` not in params this code will throw 'KeyError' so we can check both conditions

File: tests/conftest.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def mask():
 
 @pytest.fixture
 def bboxes():
-    return [[15, 12, 75, 30, 1], [55, 25, 90, 90, 2]]
+    return [[0.15, 0.12, 0.75, 0.30, 1], [0.55, 0.25, 0.90, 0.90, 2]]
```
Comment (albu) [inline tests/conftest.py]: why did you changed this?

File: tests/test_transforms.py [hunk 3]
```diff
@@ -99,7 +99,6 @@ def test_elastic_transform_interpolation(monkeypatch, interpolation):
     [A.ShiftScaleRotate, {'rotate_limit': 45}],
     [A.RandomScale, {'scale_limit': 0.5}],
     [A.RandomSizedCrop, {'min_max_height': (80, 90), 'height': 100, 'width': 100}],
-    [A.RandomSizedBBoxSafeCrop, {'height': 100, 'width': 100}],
```
Comment (Dipet) [inline tests/test_transforms.py]: Why did you remove this tests?

File: albumentations/augmentations/transforms.py (hunk 1 above)
Comment (Jae-Hyuck) [reply albumentations/augmentations/transforms.py]: https://github.com/albu/albumentations/blob/d85bab59eb8ccb0a2fec86750f94173e18e86395/albumentations/core/transforms_interface.py#L56-L58

if `bboxes` is not given, line 57 will throw 'KeyError', before enter the `get_params_dependent_on_targets` method.

File: tests/conftest.py (hunk 2 above)
Comment (Jae-Hyuck) [reply tests/conftest.py]: Interger bboxes throw errors while checking **test_serialization.py**.
RandomSizedBBoxSafeCrop only allows float bboxes now.
In my understanding, float bboxes is standard for this library, so I changed it.
Am I wrong?

File: tests/test_transforms.py (hunk 3 above)
Comment (Jae-Hyuck) [reply tests/test_transforms.py]: After this commit, `bboxes` parameter is essential to run `RandomSizedBBoxSafeCrop`.  So, I tried to add any `bboxes` parameter to the test. But there are some augmentations not supporting `bboxes` in the test, so I just removed `RandomSizedBBoxSafeCrop`. If this is not a good way, please give me an advice.

Comment (Dipet) [reply tests/test_transforms.py]: I see, this tests work only with images or images and masks. Add RandomSizedBBoxSafeCrop to this tests was a mistake.

File: tests/conftest.py (hunk 2 above)
Comment (Dipet) [reply tests/conftest.py]: @albu He is right, tests functions does not call Compose transform, so bboxes  in test are not normalized. 

I think, need rename this function to `normalized_bboxes` and return back original function. Then in `test_serialization` rename all `bboxes` test functions params to `normalized_bboxes`, except `test_transform_pipeline_serialization_with_bboxes`

File: albumentations/augmentations/transforms.py (hunk 1 above)
Comment (Dipet) [reply albumentations/augmentations/transforms.py]: @albu This transform works only with bboxes, so if we get `KeyError` it will be right behavior

File: tests/conftest.py (hunk 2 above)
Comment (albu) [reply tests/conftest.py]: or `albumentations_boxes`.

File: albumentations/augmentations/transforms.py (hunk 1 above)
Comment (albu) [reply albumentations/augmentations/transforms.py]: Then we should assert that this transform does not work without bboxes.

Comment (Jae-Hyuck) [reply albumentations/augmentations/transforms.py]: @albu Done.

File: tests/conftest.py (hunk 2 above)
Comment (Jae-Hyuck) [reply tests/conftest.py]: @Dipet Fixed.

**Comment by mirko-m** [comment]
Thanks for fixing this. It seems to me that the code in `get_params_dependent_on_targets` assumes that the bboxes are in the `albumentations` format. Is that intentional? I got an unexpected result, because I passed bboxes in a different format, but there is nothing to warn the user against doing that.

Maybe this could be fixed with a call to `convert_bboxes_to_albumentations` ?
